addons/web/static/src/views/fields/field.js

```diff
@@ -162,33 +162,41 @@ export class Field extends Component {
 Field.template = "web.Field";
 
 Field.parseFieldNode = function (node, models, modelName, viewType, jsClass) {
-    const name = node.getAttribute("name");
-    const widget = node.getAttribute("widget");
-    const fields = models[modelName];
-    const modifiers = JSON.parse(node.getAttribute("modifiers") || "{}");
-    const field = getFieldFromRegistry(fields[name].type, widget, viewType, jsClass);
     const fieldInfo = {
-        name,
         viewType,
-        widget,
-        modifiers,
-        field,
-        context: node.getAttribute("context") || "{}",
-        string: node.getAttribute("string") || fields[name].string,
-        help: node.getAttribute("help"),
-        onChange: archParseBoolean(node.getAttribute("on_change")),
-        forceSave: archParseBoolean(node.getAttribute("force_save")),
-        options: evaluateExpr(node.getAttribute("options") || "{}"),
-        alwaysInvisible: modifiers.invisible === true || modifiers.column_invisible === true,
-        decorations: {}, // populated below
-        attrs: {}, // populated below
+        modifiers: {},
+        context: "{}",
+        options: {},
+        alwaysInvisible: false,
+        decorations: {},
+        attrs: {},
     };
-    if (node.getAttribute("domain")) {
-        // TODO WOWl: remove with new model?
-        fieldInfo.domain = node.getAttribute("domain");
-    }
+
     for (const { name, value } of node.attributes) {
-        if (name.startsWith("decoration-")) {
+        if (name === "name") {
+            fieldInfo.name = value;
+        } else if (name === "widget") {
+            fieldInfo.widget = value;
+        } else if (name === "modifiers") {
+            fieldInfo.modifiers = JSON.parse(value);
+            fieldInfo.alwaysInvisible =
+                fieldInfo.modifiers.invisible === true ||
+                fieldInfo.modifiers.column_invisible === true;
+        } else if (name === "context") {
+            fieldInfo.context = value;
+        } else if (name === "string") {
+            fieldInfo.string = value;
+        } else if (name === "help") {
+            fieldInfo.help = value;
+        } else if (name === "on_change") {
+            fieldInfo.onChange = archParseBoolean(value);
+        } else if (name === "options") {
+            fieldInfo.options = evaluateExpr(value);
+        } else if (name === "force_save") {
+            fieldInfo.forceSave = archParseBoolean(value);
+        } else if (name === "domain") {
+            fieldInfo.domain = value;
+        } else if (name.startsWith("decoration-")) {
```

Comment:
if (["name", "context",  ... ].includes(name) {
   fieldInfo[name] = value
} ...

Reply:
Just an idea to reduce the code

Reply:
That's a good idea.